Add pod_push_trunk action

squarefrog opened the pull request:
This action allows you to push a Podspec to trunk.

Supplying a `path` argument passes the path to `pod trunk`

For example:

``` ruby
# runs pod trunk push
pod_push_trunk

#runs pod trunk push 'Example/Example.podspec'
pod_push_trunk(path: 'Example/Example.podspec')
```
# 

I'm not entirely happy with the `available_options` and spec here. I feel it would make more sense to check if the path:
1. Matches `/.podspec$/`
2. File exists

As a result the current spec for pushing with an argument looks for an irrelevant file. I'd appreciate some help fixing this spec if possible!

squarefrog committed:
Add pod_push_trunk action

This action allows you to push a podspec to trunk.

squarefrog commented on lib/fastlane/actions/pod_push_trunk.rb:
```diff
@@ -0,0 +1,54 @@
+                                       description: "The Podspec you want to push",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         raise "Couldn't find file at path '#{value}'".red unless File.exist?(value)
```
This block should do two things:

1. Check if the path matches `\.podspec$\`
2. Check if the file exists

KrauseFx replied:
How about
```ruby
verify_block: proc do |value|
  raise "Couldn't find file at path '#{value}'".red unless File.exist?(value)
  raise "File must be a `.podspec`'".red unless value.end_with?(".podspec")
end
```

squarefrog commented on spec/actions_specs/pod_push_trunk_spec.rb:
```diff
@@ -0,0 +1,21 @@
+          pod_push_trunk(path: './fastlane/README.md')
+        end").runner.execute(:test)
+
+        expect(result).to eq("pod trunk push './fastlane/README.md'")
```
This should check an actual podspec. Is there a way to create a temporary file somewhere, run the spec, then remove it after?

KrauseFx replied:
You could do that, or you could add a file to the `spec/fixtures` folder

KrauseFx commented on docs/Actions.md:
```diff
@@ -1348,6 +1348,18 @@ This will find the first podspec in the folder. You can also pass in the specifi
+pod_push_trunk
+
+# Alternatively, supply the Podspec file path
+pod_push_trunk(path: 'TSMessages.podspec')
```
:heart_eyes:

KrauseFx commented on lib/fastlane/actions/pod_push_trunk.rb:
```diff
@@ -0,0 +1,54 @@
+        end
+
+        result = Actions.sh("#{command}")
+        Helper.log.info "Successfully pushed Podspec".green
```
Missing nice emoji here :stuck_out_tongue_winking_eye:

KrauseFx commented:
Please add a comment when this is corrected, so that I get a notification :+1:

KrauseFx commented:
Thanks for the pull request @squarefrog

squarefrog committed:
Finalise pod_push_trunk behaviour

squarefrog commented:
@KrauseFx OK I addressed the above issues. 3 questions:

1. Do you want me to rebase/squash this into a single commit?
2. Do I need to write specs for the config options?
3. I named this `pod_push_trunk` as there are two methods of pushing:

    $ pod trunk push
    $ pod repo push

Do you feel the behaviour between the two is distinct enough or should there be one generic:

    # Push to a private repo
    pod_push(path: "A.podspec", repo: "MyPrivateRepo")

    # Push to trunk
    pod_push(path: "A.podspec")

KrauseFx commented:
> Do you want me to rebase/squash this into a single commit?
No, that's fine.
> Do I need to write specs for the config options?
No, tests for the config system are in `fastlane_core` :+1: 
> I named this pod_push_trunk as there are two methods of pushing:
It's good in one action! :+1:

KrauseFx commented:
Looks great, thanks @squarefrog for your contribution! :smile:

squarefrog commented:
> It's good in one action! :+1:

Sorry do you mean create a separate action, or add the functionality into `pod_trunk_push`?

KrauseFx commented:
Having everything in one action is good. It's already live btw :+1: Thanks again
